drivers/usb_c/tcpc/tcpci.c

```diff
@@ -0,0 +1,368 @@
+		.name = "TC_REV",
+		.size = 2,
+	},
+	{
+		.addr = TCPC_REG_PD_REV,
+		.name = "PD_REV",
+		.size = 2,
+	},
+	{
+		.addr = TCPC_REG_PD_INT_REV,
+		.name = "PD_INT_REV",
+		.size = 2,
+	},
+	{
+		.addr = TCPC_REG_ALERT,
+		.name = "ALERT",
+		.size = 2,
+	},
+	{
+		.addr = TCPC_REG_ALERT_MASK,
+		.name = "ALERT_MASK",
+		.size = 2,
+	},
+	{
+		.addr = TCPC_REG_POWER_STATUS_MASK,
+		.name = "POWER_STATUS_MASK",
+		.size = 1,
+	},
+	{
+		.addr = TCPC_REG_FAULT_STATUS_MASK,
+		.name = "FAULT_STATUS_MASK",
+		.size = 1,
+	},
+	{
+		.addr = TCPC_REG_EXT_STATUS_MASK,
+		.name = "EXT_STATUS_MASK",
+		.size = 1,
+	},
+	{
+		.addr = TCPC_REG_ALERT_EXTENDED_MASK,
+		.name = "ALERT_EXTENDED_MASK",
+		.size = 1,
+	},
+	{
+		.addr = TCPC_REG_CONFIG_STD_OUTPUT,
+		.name = "CONFIG_STD_OUTPUT",
+		.size = 1,
+	},
+	{
+		.addr = TCPC_REG_TCPC_CTRL,
+		.name = "TCPC_CTRL",
+		.size = 1,
+	},
+	{
+		.addr = TCPC_REG_ROLE_CTRL,
+		.name = "ROLE_CTRL",
+		.size = 1,
+	},
+	{
+		.addr = TCPC_REG_FAULT_CTRL,
+		.name = "FAULT_CTRL",
+		.size = 1,
+	},
+	{
+		.addr = TCPC_REG_POWER_CTRL,
+		.name = "POWER_CTRL",
+		.size = 1,
+	},
+	{
+		.addr = TCPC_REG_CC_STATUS,
+		.name = "CC_STATUS",
+		.size = 1,
+	},
+	{
+		.addr = TCPC_REG_POWER_STATUS,
+		.name = "POWER_STATUS",
+		.size = 1,
+	},
+	{
+		.addr = TCPC_REG_FAULT_STATUS,
+		.name = "FAULT_STATUS",
+		.size = 1,
+	},
+	{
+		.addr = TCPC_REG_EXT_STATUS,
+		.name = "EXT_STATUS",
+		.size = 1,
+	},
+	{
+		.addr = TCPC_REG_ALERT_EXT,
+		.name = "ALERT_EXT",
+		.size = 1,
+	},
+	{
+		.addr = TCPC_REG_DEV_CAP_1,
+		.name = "DEV_CAP_1",
+		.size = 2,
+	},
+	{
+		.addr = TCPC_REG_DEV_CAP_2,
+		.name = "DEV_CAP_2",
+		.size = 2,
+	},
+	{
+		.addr = TCPC_REG_STD_INPUT_CAP,
+		.name = "STD_INPUT_CAP",
+		.size = 1,
+	},
+	{
+		.addr = TCPC_REG_STD_OUTPUT_CAP,
+		.name = "STD_OUTPUT_CAP",
+		.size = 1,
+	},
+	{
+		.addr = TCPC_REG_CONFIG_EXT_1,
+		.name = "CONFIG_EXT_1",
+		.size = 1,
+	},
+	{
+		.addr = TCPC_REG_MSG_HDR_INFO,
+		.name = "MSG_HDR_INFO",
+		.size = 1,
+	},
+	{
+		.addr = TCPC_REG_RX_DETECT,
+		.name = "RX_DETECT",
+		.size = 1,
+	},
+	{
+		.addr = TCPC_REG_RX_BYTE_CNT,
+		.name = "RX_BYTE_CNT",
+		.size = 1,
+	},
+	{
+		.addr = TCPC_REG_RX_BUF_FRAME_TYPE,
+		.name = "RX_BUF_FRAME_TYPE",
+		.size = 1,
+	},
+	{
+		.addr = TCPC_REG_TRANSMIT,
+		.name = "TRANSMIT",
+		.size = 1,
+	},
+	{
+		.addr = TCPC_REG_VBUS_VOLTAGE,
+		.name = "VBUS_VOLTAGE",
+		.size = 2,
+	},
+	{
+		.addr = TCPC_REG_VBUS_SINK_DISCONNECT_THRESH,
+		.name = "VBUS_SINK_DISCONNECT_THRESH",
+		.size = 2,
+	},
+	{
+		.addr = TCPC_REG_VBUS_STOP_DISCHARGE_THRESH,
+		.name = "VBUS_STOP_DISCHARGE_THRESH",
+		.size = 2,
+	},
+	{
+		.addr = TCPC_REG_VBUS_VOLTAGE_ALARM_HI_CFG,
+		.name = "VBUS_VOLTAGE_ALARM_HI_CFG",
+		.size = 2,
+	},
+	{
+		.addr = TCPC_REG_VBUS_VOLTAGE_ALARM_LO_CFG,
+		.name = "VBUS_VOLTAGE_ALARM_LO_CFG",
+		.size = 2,
+	},
+};
+
+int tcpci_read_reg8(const struct i2c_dt_spec *i2c, uint8_t reg, uint8_t *value)
```

Comment:
Add comment @brief and the same for the other function.

Reply:
Is there a need to add functions documentation to source files if header or API header contains them? I see that the ucpd driver contains documentation for its implementation functions duplicated from the _usbc_tcpc.h_. And the drivers functions should be compliant to the API functions and their documentation.

Reply:
I understand, but I don't know if it's convention to have a brief explanation of the function here too. Other reviewers, please comment if it's okay to just document the functions in the header file.